docs/doc_sources/user_guides/environment_variables.rst

```diff
@@ -50,3 +50,61 @@ The value of the variable is a bit-mask, with the following supported values:
       - Enables tracing of PI calls
     * - ``-1``
       - Enables all levels of tracing
+
+Variable ``ZE_FLAT_DEVICE_HIERARCHY``
```

Comment:
this should be cross-linked to `get_composite_devices` and other composite device related functions

the docstrings in those functions should also refer to here